lib/dotcom/utils/date_time.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Dotcom.Utils.DateTime do
+  @moduledoc """
+  A collection of functions for working with date_times.
+
+  Consuming modules are responsible for parsing or converting date_times.
+  They should *always* call `coerce_ambiguous_time/1` before using a date_time.
+  This is mainly because Timex has so many functions that serve as entry points to date_times.
+  Those functions can return ambiguous date_times during DST transitions.
+  """
+
+  require Logger
+
+  use Timex
+
+  @typedoc """
+  A range of time that can be open ended on either side.
+  """
+  @type time_range() :: {DateTime.t() | nil, DateTime.t() | nil}
+
+  @timezone Application.compile_env!(:dotcom, :timezone)
+
+  @doc """
+  Returns the timezone for the application.
+  """
+  def timezone(), do: @timezone
+
+  @doc """
+  Get the date_time in the set @timezone.
+  """
+  @spec now() :: DateTime.t()
+  def now(), do: Timex.now(@timezone)
+
+  @doc """
+  Timex can give us ambiguous times during DST transitions; we choose the later time.
+  In the *very* rare case that we are given an {:error, _} tuple, we default to now.
+  In the even rarer case that we are given some other argument type, we also default to now.
```

Comment:
`coerce_ambiguous_time/1` will get called with all sorts of times, not just `current_time` or `Timex.now()` or equivalent, right?

I'm nervous about what looks like _almost_ a silent failure along with potentially doing something wrong, rather than just crashing and surfacing an error to the user.

Reply:
That's why we log it and *don't* crash the app because it should only be happening in very strange situations.

Reply:
Yeah but then the rider experience is that we're confidently telling them something that's not true.

Example: The last trip of the day for a given route ends at a time that's ambiguous and would lead to this error. We discover the error, but instead of telling the rider that there's an unusual error and we don't know when the last trip of the day is, we tell them confidently that the last trip of the day is _now_ (whatever now is). 

IMO, that's a worse experience than a 500.

Reply:
this is an insanely rare case. so rare that I'm not sure how to even trigger it. the way we handle these kinds of issues elsewhere is to log the argument and return some kind of reasonable value. this is not a decision that was made recently as we had multiple instances of the app crashing. this keeps in line with what we've done elsewhere.